consensus/consensus-types/src/common.rs

```diff
@@ -64,61 +78,79 @@ impl Payload {
                 .with_min_len(100)
                 .map(|txn| txn.raw_txn_bytes_len())
                 .sum(),
+            Payload::InQuorumStore(_) => 0, // quorum store TODO
+            Payload::Empty => 0,
         }
     }
-}
-
-// TODO: What I really want is an iterator that isn't necessarily a vector (e.g., read lazily from RocksDB). This doesn't seem like the way.
-impl IntoIterator for Payload {
-    type Item = SignedTransaction;
-    type IntoIter = std::vec::IntoIter<Self::Item>;
 
-    fn into_iter(self) -> Self::IntoIter {
+    pub fn verify(&self, validator: &ValidatorVerifier) -> anyhow::Result<()> {
         match self {
-            Payload::DirectMempool(txns) => txns.into_iter(),
+            Payload::Empty => Ok(()),
+            Payload::DirectMempool(_) => Ok(()),
+            Payload::InQuorumStore(proofs) => {
+                for proof in proofs.iter() {
+                    proof.verify(validator)?;
+                }
+                Ok(())
+            }
         }
     }
 }
-
 impl fmt::Display for Payload {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
             Payload::DirectMempool(txns) => {
                 write!(f, "InMemory txns: {}", txns.len())
             }
+            Payload::InQuorumStore(proofs) => {
+                write!(f, "InMemory poavs: {}", proofs.len())
```

Comment:
Lets be consistent. Use `pos`; or `poav` everywhere else if that's preferrred

Reply:
Agree.